src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            //find out if the autologon was already enabled, show warning in that case
+            ShowAutoLogonWarningIfAlreadyEnabled(regHelper, userName);
+
+            var windowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            windowsHelper.SetAutoLogonPassword(logonPassword);
+
+            regHelper.UpdateAutoLogonSettings(userName, domainName);
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(AutoLogonRegistryManager regHelper, string userName)
+        {
+            regHelper.FetchAutoLogonUserDetails(out string autoLogonUserName, out string domainName);
+            if(autoLogonUserName != null && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                _terminal.WriteLine(string.Format(StringUtil.Loc("AutoLogonAlreadyEnabledWarning"), userName));
+            }
+        }
+
+        private bool IsCurrentUserSameAsAutoLogonUser()
+        {
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>());            
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if(string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException("AutoLogon is not configured on the machine.");
+            }
+
+            var nativeWindowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            return nativeWindowsHelper.HasActiveSession(domainName, userName);
+        }
+
+        private void DisplayWarningsIfAny(AutoLogonRegistryManager regHelper)
+        {
+            var warningReasons = regHelper.GetAutoLogonRelatedWarningsIfAny();
+            if(warningReasons.Count > 0)
+            {
+                WriteSection(StringUtil.Loc("UITestingWarning"));
+                for(int i=0; i < warningReasons.Count; i++)
+                {
+                    _terminal.WriteLine(String.Format("{0} - {1}", (i+1).ToString(), warningReasons[i]));
+                }
+            }
+        }
+
+        private void ConfigurePowerOptions()
+        {
+            var whichUtil = HostContext.GetService<IWhichUtil>();
+            var filePath = whichUtil.Which("powercfg.exe");
+            string[] commands = new string[] {"/Change monitor-timeout-ac 0", "/Change monitor-timeout-dc 0"};
+
+            foreach (var command in commands)
+            {
+                try
+                {
+                    Trace.Info($"Running powercfg.exe with {command}");
+                    using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                    {
+                        processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                        {
+                            Trace.Verbose(message.Data);
```

Comment:
Ting, should such internals be documented somewhere? Based on my exp, I have seen verbose logging available on shipped bits as well. 

---
In reply to: [118022049](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118022049) [](ancestors = 118022049)